cmd/sessionstorage.go

```diff
@@ -0,0 +1,216 @@
+	flags.StringVar(&options.sortkey, "sortkey", "expires", "sort by specific object property")
+	flags.StringVar(&options.sortdir, "sortdir", "ASC", "sort direction, ASC or DESC")
+
+	return cmd
+}
+
+func sessionsShow(options sessionStorageOptions) error {
+	api := authApi.New(curl())
+	if options.userID != "" {
+		userSessions, err := api.UserSessions(options.offset, options.limit,
+			options.sortkey, options.sortdir, options.userID)
+		if err != nil {
+			return err
+		}
+		return stdout(userSessions)
+	} else if options.sourceID != "" {
+		sourceSessions, err := api.SourceSessions(options.offset, options.limit,
+			options.sortkey, options.sortdir, options.sourceID)
+		if err != nil {
+			return err
+		}
+		return stdout(sourceSessions)
+	}
+	return nil
+}
+
+//
+//
+func sessionsSearchCmd() *cobra.Command {
+	options := sessionStorageOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "search",
+		Short: "Search for sessions",
+		Long:  `Search for sessions. Search with keywords, userid or type`,
+		Example: `
+	privx-cli sessions search [access flags] JSON-FILE
+		`,
+		Args:         cobra.ExactArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return sessionsSearch(options, args)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.IntVar(&options.offset, "offset", 0, "where to start fetching the items")
+	flags.IntVar(&options.limit, "limit", 50, "number of items to return")
+	flags.StringVar(&options.sortkey, "sortkey", "expires", "sort by specific object property")
+	flags.StringVar(&options.sortdir, "sortdir", "ASC", "sort direction, ASC or DESC")
+
+	return cmd
+}
+
+func sessionsSearch(options sessionStorageOptions, args []string) error {
+	var searchParams authApi.SearchParams
+	api := authApi.New(curl())
+
+	err := decodeJSON(args[0], &searchParams)
+	if err != nil {
+		return err
+	}
+
+	sessions, err := api.SearchSessions(options.offset, options.limit,
+		options.sortkey, options.sortdir, &searchParams)
+	if err != nil {
+		return err
+	}
+
+	return stdout(sessions)
+}
+
+//
+//
+func terminateSessionCmd() *cobra.Command {
+	options := sessionStorageOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "terminate",
+		Short: "Terminate single session by ID",
+		Long:  `Terminate single session by sessionID`,
+		Example: `
+	privx-cli sessions terminate [access flags] --id <SESSION-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return terminateSession(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.sessionID, "id", "", "Session ID")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func terminateSession(options sessionStorageOptions) error {
+	api := authApi.New(curl())
+
+	return api.TerminateSession(options.sessionID)
+}
+
+//
+//
+func terminateUserSessionsCmd() *cobra.Command {
+	options := sessionStorageOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "terminate-user",
```

Comment:
Should we maybe rename this command?
We could call it for example "terminate-all" which lines up with the "terminate" sub command, and we mention already that those sessions are user sessions so maybe the "user" inside the command is not required at all.
Do you have any thoughts on that?

Reply:
i resorted to terminate-user to terminate all sessions for a user because I already used the terminal sub command for termination of a single session by session-id.
One option I guess we have is we could potentially have a single terminate command but based on flags (session-id or user-id), terminate sessions.
But then I dont know if it could be a bit misleading in a sense that terminate by session id terminates a single session, but terminate by userid terminates all session for the user-id?

Reply:
but, it does sound weird - terminate-user :dagger:

Reply:
I will switch it to terminate-all